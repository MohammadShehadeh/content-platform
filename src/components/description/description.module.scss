.description {
  line-height: 1.5;
  margin-bottom: var(--space-sm);
}

// Size variants
.sm {
  font-size: var(--font-size-xs);
}

.md {
  font-size: var(--font-size-sm);
}

.lg {
  font-size: var(--font-size-md);
}

// Color variants
.default {
  color: hsl(var(--foreground));
}

.muted {
  color: hsl(var(--muted-foreground));
}

.disableGutter {
  margin-bottom: 0;
}
