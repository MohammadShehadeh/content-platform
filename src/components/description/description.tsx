import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import styles from './description.module.scss';

const descriptionVariants = cva(styles.description, {
  variants: {
    size: {
      sm: styles.sm,
      md: styles.md,
      lg: styles.lg,
    },
    color: {
      default: styles.default,
      muted: styles.muted,
    },
    disableGutter: {
      true: styles.disableGutter,
    },
  },
  defaultVariants: {
    size: 'md',
    color: 'default',
    disableGutter: false,
  },
});

interface DescriptionProps
  extends Omit<React.HTMLAttributes<HTMLParagraphElement>, 'color'>,
    VariantProps<typeof descriptionVariants> {
  asChild?: boolean;
  disableGutter?: boolean;
}

export const Description = ({
  children,
  className,
  asChild = false,
  disableGutter,
  size,
  color = 'default',
  ...props
}: DescriptionProps) => {
  const Comp = asChild ? Slot : 'p';

  return (
    <Comp
      className={descriptionVariants({ size, color, className, disableGutter })}
      {...props}
    >
      {children}
    </Comp>
  );
};
