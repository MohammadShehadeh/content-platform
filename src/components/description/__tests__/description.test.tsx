import { render, screen } from '@testing-library/react';

import { Description } from '@/components/description';

describe('Description', () => {
  it('should render children correctly', () => {
    render(<Description>Hello World</Description>);
    expect(screen.getByText('Hello World')).toBeInTheDocument();
  });

  it('should apply size variants correctly', () => {
    const { rerender } = render(<Description size="sm">Small</Description>);
    expect(screen.getByText('Small')).toHaveClass('sm');

    rerender(<Description size="lg">Large</Description>);
    expect(screen.getByText('Large')).toHaveClass('lg');
  });

  it('should apply color variants correctly', () => {
    const { rerender } = render(<Description color="default">Default</Description>);
    expect(screen.getByText('Default')).toHaveClass('default');

    rerender(<Description color="muted">Muted</Description>);
    expect(screen.getByText('Muted')).toHaveClass('muted');
  });

  it('should apply disableGutter prop correctly', () => {
    render(<Description disableGutter>No Gutter</Description>);
    expect(screen.getByText('No Gutter')).toHaveClass('disableGutter');
  });

  it('should apply custom className', () => {
    render(<Description className="custom-description">Custom Class</Description>);
    expect(screen.getByText('Custom Class')).toHaveClass('custom-description');
  });

  it('should render as child component when asChild is true', () => {
    render(
      <Description asChild>
        <span data-testid="child-span">As Child</span>
      </Description>
    );
    expect(screen.getByTestId('child-span')).toBeInTheDocument();
    expect(screen.getByText('As Child')).toHaveClass('description');
  });

  it('should match snapshot', () => {
    const { container } = render(<Description>Hello</Description>);
    expect(container).toMatchSnapshot();
  });
});
