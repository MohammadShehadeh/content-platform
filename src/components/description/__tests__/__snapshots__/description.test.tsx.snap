// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description matches snapshot 1`] = `
<div>
  <p
    class="description md default"
  >
    Hello
  </p>
</div>
`;

exports[`Description should match snapshot 1`] = `
<div>
  <p
    class="description md default"
  >
    Hello
  </p>
</div>
`;
